packages/react-components/react-charts-preview/library/src/components/CommonComponents/Popover.tsx

```diff
@@ -0,0 +1,247 @@
+      <Popover
+        positioning={{ target: virtualElement, autoSize: 'always', offset: 20, coverTarget: false }}
+        open={props.isPopoverOpen}
+        inline
+      >
+        <PopoverSurface>
+          {/** Given custom callout, then it will render */}
+          {props.customCallout && props.customCallout.customizedCallout && props.customCallout.customizedCallout}
+          {/** single x point its corresponding y points of all the bars/lines in chart will render in callout */}
+          {(!props.customCallout || !props.customCallout.customizedCallout) &&
+            props.isCalloutForStack &&
+            _multiValueCallout()}
+          {/** single x point its corresponding y point of single line/bar in the chart will render in callout */}
+          {(!props.customCallout || !props.customCallout.customizedCallout) && !props.isCalloutForStack && (
+            <div className={classes.calloutContentRoot}>
+              <div className={classes.calloutDateTimeContainer}>
+                <div className={classes.calloutContentX}>{props.XValue} </div>
+                {/*TO DO  if we add time for callout then will use this */}
+                {/* <div className={classNames.calloutContentX}>07:00am</div> */}
+              </div>
+              <div
+                className={classes.calloutInfoContainer}
+                style={{
+                  ...(props.ratio && {
+                    display: 'flex',
+                    alignItems: 'flex-end',
+                  }),
+                  borderLeft: `4px solid ${props.color}`,
+                }}
+              >
+                <div className={classes.calloutBlockContainer}>
+                  <div className={classes.calloutlegendText}>{convertToLocaleString(Legend, props.culture)}</div>
+                  <div
+                    className={classes.calloutContentY}
+                    style={{ color: props.color ? props.color : tokens.colorNeutralForeground1 }}
+                  >
+                    {convertToLocaleString(YValue, props.culture)}
+                  </div>
+                </div>
+                {!!props.ratio && (
+                  <div className={classes.ratio}>
+                    <>
+                      <span className={classes.numerator}>{convertToLocaleString(props.ratio[0], props.culture)}</span>/
+                      <span className={classes.denominator}>
+                        {convertToLocaleString(props.ratio[1], props.culture)}
+                      </span>
+                    </>
+                  </div>
+                )}
+              </div>
+              {!!props.descriptionMessage && (
+                <div className={classes.descriptionMessage}>{props.descriptionMessage}</div>
+              )}
+            </div>
+          )}
+        </PopoverSurface>
+      </Popover>
+    </div>
+  );
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function _multiValueCallout() {
```

Comment:
In hooks, consider using the const _multiValueCallout = () => {} form for declaring functions. It avoids incorrect binding situations for this.

Reply:
The arrow dereferencing approach does not allow us to unit test our functions. We will update them once we find a way to make our unit tests work with arrow dereferencing.